### ISSUE TITLE
Require specifying encoding when using "open" (re: PEP597)

### ISSUE THREAD

**Issue by Sxderp** [wemake-services/wemake-python-styleguide#1948]
# Rule request

Inspired by https://www.python.org/dev/peps/pep-0597

The PEP is still draft, but I think WPS can implement before adoption. Even if the PEP is rejected I think this is a good idea.

Required specifying "encoding" to `open`. Even if it's just `None`.

## Thesis

Not specifying the encoding could be considered a bug. Making the effort to choose which encoding is better. At the moment there is no builtin "locale" so we have to settle for `None` or call `locale.getpreferredencoding()` all the time. The second option is a bit clumbersome.

```python
## Wrong
with open('filename.txt') as fd:
    fd.read()

## Correct
with open('filename.txt', encoding='ascii') as fd:
  fd.read()

with open('filename.txt', encoding='utf8') as fd:
  fd.read()

# Unfortunately we have to settle for "None" to indicate "locale".
with open('filename.txt', encoding=None) as fd:
  fd.read()
```

## Reasoning

Everything described in the PEP.

**Comment by sobolevn** [comment]
Great idea, I will totally do this (either here or in `typed-linter` that I am developing right now)!

Why not just implementing it here? Because `typed-linter` will support auto-fixing problems. And this looks like it can be autofixed.

**Comment by GibranHL0** [comment]
I'd like to take this issue as well, pls 😌

**Comment by GibranHL0** [comment]
#2082 Opened

**Comment by Sxderp** [comment]
https://github.com/wemake-services/wemake-python-styleguide/pull/2082#pullrequestreview-2507464782

Looks like ruff does this.
